zoo/meilisearch/views.py

```diff
@@ -1,48 +1,96 @@
-import meilisearch
 import structlog
 from django.apps import apps
-from django.conf import settings
 from django.views.generic import TemplateView
 
+from .meilli_client import meili_client
+
 log = structlog.get_logger()
 
 
 class MeiliSearchView(TemplateView):
     template_name = "search_overview.html"
     context_object_name = "context"
-    per_page = 25
+    meili_limit = 20
 
     @staticmethod
     def _objects_from_result(search_results, index, result_objects=None):
-        if index["name"] not in result_objects.keys():
-            result_objects[index["name"]] = []
         try:
             model = apps.get_model(index["uid"], index["name"])
             for result in search_results:
-                result_objects[index["name"]].append(model.objects.get(pk=result["id"]))
+                key = index["name"].lower()
+                if key not in result_objects.keys():
+                    result_objects[key] = []
+                result_objects[key].append(model.objects.get(pk=result["id"]))
         except LookupError:
             for result in search_results:
-                result_objects[index["name"]].append(result["id"])
+                result_objects[index["name"].lower()].append(result["id"])
 
         return result_objects
 
-    def _search(self, query_param):
-        meili_client = meilisearch.Client(
-            settings.MEILI_HOST, settings.MEILI_MASTER_KEY
-        )
-        indexes = meili_client.get_indexes()
+    def _search(self, search_query, index_type, offset=0, limit=meili_limit):
         objects_to_return = {}
+        new_offset = 0
+        indexes = meili_client.get_indexes()
         for index in indexes:
-            results = meili_client.get_index(index["uid"]).search(query_param)["hits"]
-            objects_for_index = self._objects_from_result(
-                results, index, objects_to_return
+            results = meili_client.get_index(index["uid"]).search(
+                query=search_query, opt_params={"offset": offset, "limit": limit}
             )
-            objects_to_return.update(objects_for_index)
-        return objects_to_return
+            objects_to_return[f"total_{index['name'].lower()}"] = results["nbHits"]
+
+            if index_type == index["uid"]:
+                new_offset = results["offset"]
+                total_hits = results["nbHits"]
```

Comment:
this should have a default... otherwise it won't be defined if the condition isn't met, so it will break on line 51